### ISSUE TITLE
Provide an unstable version

### ISSUE THREAD

**Issue by StanleySweet** [flathub/com.play0ad.zeroad#9]
It has been quite a while since the last 0 A.D. release (Still working on it, sorry about that) and I was wondering if it was possible for you guys to also provide a bleeding edge flatpack for people that would want to try the dev version?

**Comment by corecontingency** [comment]
Flathub supports a beta branch for unstable releases, but a `-git / nightly` version isn't supported yet.

As explained [here](https://blogs.gnome.org/alexl/2019/02/19/changes-in-flathub-land/):

> Exactly if and how this is used is up to each individual application maintainer, but the goal of this is to have a way for developers to get early releases of new stable versions into the hand of regular users.
> 
> This isn’t meant to be used for nightly builds, but for releases that has some level of testing and are expected to mostly work and be usable to non-developer end-users.

However, if you want to have a `-git` flatpak, you can always just publish it to a flatpak repo on 0ad's website and setup some CI to automatically rebuild on every commit. GNOME [already does this](https://wiki.gnome.org/Apps/Nightly).

Users could add your proposed nightly flatpak repo with a command like this:

`flatpak remote-add --if-not-exists 0ad-nightly https://nightly.play0ad.com/0ad-nightly.flatpakrepo`

And then install the nightly version of 0ad like this:

`flatpak install 0ad-nightly com.play0ad.zeroad.Devel`

**Comment by barthalion** [comment]
I'd say that the bottom line here is that it's ultimately up to application developers what they want to use the beta branch for. The main problem is that we (as in Flathub admins) expect manifest to point to specific commit for reproducibility and currently there's no automation for updating that regularly.

**Comment by tuxayo** [comment]
Hi :)
A lot of benefit could be achieved just with an RC channel, not a nightly. Like now: https://wildfiregames.com/forum/topic/41264-alpha-25-pre-releaserelease-candidate-build-testing/page/16/?tab=comments#comment-445469

How does that sound?

**Comment by tuxayo** [comment]
@StanleySweet is there another need in-between nighties and RCs? I don't know if mid cycle wider testing of the dev version is occasionally needed.

**Comment by na-Itms** [comment]
With our new development workflow, mock release bundles will be generated twice a month from the development version. They can be found at https://jenkins.wildfiregames.com/job/0ad-bundles/. They are probably a better candidate for unstable version than using directly the git/nightly version.

**Comment by StanleySweet** [comment]
The bimonthly versions should now be available on the beta channel.

**Comment by na-Itms** [comment]
Great work Stan thanks!

Am I correct that this will not automatically pick up the new version of bundles generated on our jenkins? Will we have to perform a manual action every two weeks to update the SHA sum of the bundle in the yaml file?

**Comment by StanleySweet** [comment]
> Great work Stan thanks!
> 
> Am I correct that this will not automatically pick up the new version of bundles generated on our jenkins? Will we have to perform a manual action every two weeks to update the SHA sum of the bundle in the yaml file?

Yeah that's right. I'll make a new PR with the new SHA every two weeks, and it should update the package once merged.

**Comment by TheShadowOfHassen** [comment]
> Yeah that's right. I'll make a new PR with the new SHA every two weeks, and it should update the package once merged.

I can do that if you have things you'd prefer to do (I'd need to be told where to get the SHA)

**Comment by StanleySweet** [comment]
That'd be nice. I currently download the files and compute it but it would be nicer if it was provided by Jenkins.

**Comment by na-Itms** [comment]
> it would be nicer if it was provided by Jenkins.

It is! You'll find them in the post-actions step. For instance https://jenkins.wildfiregames.com/job/0ad-bundles/67/pipeline-console/log?nodeId=67
